zap/src/main/java/org/zaproxy/zap/extension/search/SearchThread.java

```diff
@@ -270,6 +270,16 @@ private void search() {
                             }
                         }
                     }
+                    if (Type.Tag.equals(reqType)) {
+                        for (String tag : message.getHistoryRef().getTags()) {
+                            matcher = pattern.matcher(tag);
+                            while (matcher.matches() && !pcc.allMatchesProcessed()) {
```

Comment:
- The use of `matches` is a deviation from all other search types, I'd use `find` here too.
 - The loop is not necessary (there's only a single tag being checked in each `for` iteration), an if statement is enough.
 - Should break out of the `for` loop when all matches are processed (no point iterating them all if we are not going to add them).

Reply:
The last point was not addressed yet.

Reply:
It's still looping them if all matches were already processed, hmm, my initial suggestion is also not the best behaviour, adding the check to the if of the Tag type check is though.

Reply:
Move `!pcc.allMatchesProcessed()` to right hand of `Type.Tag.equals(reqType)`.